packages/env-tests/src/tests/oracle.ts

```diff
@@ -0,0 +1,60 @@
+import { CeloContract } from '@celo/contractkit'
+import { describe, expect, test } from '@jest/globals'
+import BigNumber from 'bignumber.js'
+import { Context } from '../context'
+import { fundAccount, getKey, ONE, TestAccounts } from '../scaffold'
+
+export function runOracleTest(context: Context) {
+  describe('Oracle Test', () => {
+    const logger = context.logger.child({ test: 'exchange' })
+    beforeAll(async () => {
+      await fundAccount(context, TestAccounts.Exchange, ONE.times(2))
+    })
+
+    // TODO: Check if oracle account is authorized
+    test('report a rate', async () => {
+      const from = await getKey(context.mnemonic, TestAccounts.Oracle)
+      context.kit.addAccount(from.privateKey)
+      context.kit.defaultAccount = from.address
+      const stableToken = await context.kit.contracts.getStableToken()
+      context.kit.defaultFeeCurrency = stableToken.address
+
+      const oracles = await context.kit.contracts.getSortedOracles()
+
+      const isOracle = await oracles.isOracle(CeloContract.StableToken, from.address)
+
+      expect(isOracle).toBeTruthy()
+
+      const oracleRates = await oracles.getReports(CeloContract.StableToken)
+      const ourRate = oracleRates.find((_) => _.address === from.address)
+
+      let rateToReport: BigNumber
+      if (!ourRate) {
+        const currentMedianRate = await oracles.medianRate(CeloContract.StableToken)
+        rateToReport = currentMedianRate.rate
+        logger.debug(
+          {
+            rate: currentMedianRate.rate.toString(),
+          },
+          'no existing rate, using the median'
+        )
+      } else {
+        rateToReport = ourRate.rate
+        logger.debug({ rate: ourRate.rate.toString() }, 'fetched existing oracle report')
+      }
+
+      // Move the rate in one direction or another
+      rateToReport = Math.random() < 0.5 ? rateToReport.times(1.05) : rateToReport.times(0.95)
+
+      const reportTx = await oracles.report(CeloContract.StableToken, rateToReport, from.address)
+      const reportTxReceipt = await reportTx.sendAndWaitForReceipt({ from: from.address })
+      logger.debug({ receipt: reportTxReceipt }, 'rate reported')
+
+      const newOracleRates = await oracles.getReports(CeloContract.StableToken)
+      const ourNewRate = newOracleRates.find((_) => _.address === from.address)
```

Comment:
this seems pretty obvious
should we add a `getOracleRate(oracle, token)` to wrapper?